packages/astro/src/core/errors/userError.ts

```diff
@@ -0,0 +1,20 @@
+import { AstroError } from './errors.js';
+
+export interface CreateAstroError {
+	/**
+	 * The cause of the error.
+	 */
+	message: string;
+	/**
+	 * An optional message that explain the user how they could fix the error.
+	 */
+	hint?: string;
+}
+
+export function createAstroError({ message, hint }: CreateAstroError): AstroError {
+	return new AstroError({
+		name: 'AstroError',
+		message,
+		hint,
+	});
+}
```

Comment:
Maybe we should export the `AstroError` class directly? It could be used to `extend` and checked for `instanceof` by users, and errors are usually created like `throw new AstroError()` in JS.

Reply:
If you check the linear issue related to this implementation, Erika suggested avoiding exposing `AstroError` because it contains a lot of internal things that might not be needed. 

Do you still think it's worth exposing the whole `AstroError`?

Reply:
Thanks for the insights. I think it makes sense to limit the API surface if we want to be cautious, but if we're already returning an instance of `AstroError` that already exposes a bit of API 🤔 

For example, this PR limits to `message` and `hint` only, but someone could also do:

```js
const error = createAstroError({ ... })
error.setLocation(...)
```

From Erika's suggestion in Linear, maybe we should extend from `Error` only instead and not use `AstroError`?